07-eval/client_eval_test.py

```diff
@@ -50,20 +34,29 @@ async def evaluate_metrics(metrics, test_case, actual_output):
 async def test_chat_eval(traced_test):
     actual_output = OpenAIClient().chat(message)
 
-    test_case = LLMTestCase(
-        input=message,
-        actual_output=actual_output,
-        context=["Atlantic Ocean"],
+    test_case = pd.DataFrame(
+        {
+            "input": [message],
+            "output": [actual_output],
+            "reference": ["Atlantic Ocean"],
+        }
     )
 
-    # Avoid the gpt-4o default, as it is too strict on HallucinationMetric
-    eval_llm, _ = initialize_model(eval_model)
-    metrics = [
-        AnswerRelevancyMetric(model=eval_llm, threshold=0.7),
-        HallucinationMetric(model=eval_llm, threshold=0.8),
-    ]
+    eval_model = OpenAIModel(
+        model=os.getenv("EVAL_MODEL", "o4-mini"), temperature=0.0
+    )
 
-    failures = await evaluate_metrics(metrics, test_case, actual_output)
-    assert not failures, "{} metrics failed:\n".format(
-        len(failures)
-    ) + "\n".join(failures)
+    qa_eval, hallucination_eval = run_evals(
+        dataframe=test_case,
+        evaluators=[
+            QAEvaluator(eval_model),
+            HallucinationEvaluator(eval_model),
+        ],
+        provide_explanation=True,
+    )
+    with assert_all():
```

Comment:
Oh another point - it's pretty rare for both of these to pass. I think it's because the phoenix prompt templates seem to be truly binary 1 or 0, not a range that could have a custom threshold. We could use custom prompt but it complicates the example - could be good to try it out a few times to see the explanations and see if it's just because the test input/output is too small

Reply:
@mikeldking lets think about this post merge, in case you have advice